tests/PySys/misc_features/custom_init_system/run.py

```diff
@@ -0,0 +1,87 @@
+from pysys.basetest import BaseTest
+import os
+
+"""
+Validate tedge connect/disconnect use a given init system config
+
+Given a configured system
+Given a dummy init system executable, that prints out to /etc/tedge/dummy_init.out
+Given an init system config to /etc/tedge
+When we run tedge connect c8y
+When we run tedge disconnect c8y
+Then we find no error is recorded on the output of tedge connect/disconnect, also the dummy init system executable.
+Cleanup the system config file, the dummy init system executable, and the output of the dummy init system executable.
+"""
+
+
+class CustomInitSystem(BaseTest):
+    def setup(self):
+        self.tedge = "/usr/bin/tedge"
+        self.sudo = "/usr/bin/sudo"
+        self.system_conf = "/etc/tedge/system.toml"
+        self.dummy_init = "/etc/tedge/dummy_init.sh"
+        self.dummy_init_output = "/etc/tedge/dummy_init.out"
+
+        test_dir = os.getcwd() + "/misc_features/custom_init_system"
+
+        # Copy system.toml file to /etc/tedge
+        copy_system_config_file = self.startProcess(
+            command=self.sudo,
+            arguments=["cp", test_dir + "/system.toml", self.system_conf],
+            stdouterr="copy_system_config",
+        )
+
+        # Copy dummy init system file to /etc/tedge
+        copy_dummy_init_file = self.startProcess(
+            command=self.sudo,
+            arguments=["cp", test_dir + "/dummy_init.sh", self.dummy_init],
+            stdouterr="copy_dummy_init",
+        )
+
+        self.addCleanupFunction(self.custom_cleanup)
+
+    def execute(self):
```

Comment:
What about calling all the other commands from dummy_init.sh here, too?

Reply:
I don't think we need to test the `dummy_init.sh` itself.